### ISSUE TITLE
Create a User Interface with CAP (SAP HANA Cloud)

### ISSUE THREAD

**Issue by binpath** [SAPDocuments/Issues#282]
Tutorials: https://developers.sap.com/tutorials/hana-cloud-cap-create-ui.html
--------------------------

HI, can you add in the mta.yml file in the module of the ui_deployer

`    build-parameters:      
          ignore:        - '.*/'`
          
and also in the resources the  
`- name: <approuter>_html_repo_host
    type: org.cloudfoundry.managed-service
    parameters:      
        service: html5-apps-repo
        **config:        
            sizeLimit: 24**      
        service-plan: app-host`
        
  With these parameters the development space doesn't fail at the moment of `mbt build -t ./` 
  
  kind regards, 
binpath

**Comment by jung-thomas** [comment]
I'm preparing a new/separate tutorial with just the steps for deployment.  It is finished but waiting to be moved to production. However with Easter/Passover this week, many colleagues are away and it might not get setup in production until early next week.

**Comment by binpath** [comment]
@jung-thomas The tutorial is not there anymore, can you please put it back again?

**Comment by jung-thomas** [comment]
No idea what happened. The tutorial was there yesterday.  I can still see it in GitHub.  There must be an issue in the tutorial site itself. I've sent a problem report to the team responsible for the site.

**Comment by binpath** [comment]
@jung-thomas Thank you again! you are the best!

**Comment by jung-thomas** [comment]
Seems to be back now but I don't know what the root of the issue was.  
https://developers.sap.com/tutorials/hana-cloud-cap-deploy-mta.html

**Comment by binpath** [comment]
@jung-thomas thanks you! i'm trying to figure it out my parameters, know the issue that i'm having is with the development space filesystem. Every time I build I receive the 95%  of disk usage. So i'm downloading with git and build it locally. 

kind regards, 
@binpath

**Comment by jung-thomas** [comment]
The disk usage is because the build process is including it's own temporary files into the build process.  This recursively keeps adding itself to the output until you eventually fill the file system.  This is why you need to add the build-parameters ignore with at least "..*/" to filter out the build's own content. Any module (like srv or the UI deployer) that targets the project root needs this build-parameters ignore option. The mbt team is aware of this issue and is working to address it in a future version.  Until then this parameter avoids the issue.

**Comment by jung-thomas** [comment]
Although if you are just talking about the popup in the Business Application studio about the dev space being full, this can also just happen naturally if you have a project or two in the dev space. I think the total limit for a dev space is 2Gb.  That's pretty easy to hit with only two CAP projects with full node_modules folders. When I was doing the build myself in BAS I was getting this warning that I was over 80% of my dev space capacity.  I found I had two other projects in the same dev space where I deleted the node_modules folders to free up space as well.

**Comment by binpath** [comment]
@jung-thomas yes you are right, with two cap projects is enough to run out of filesystem. I remove the projects and later if a needed I clone it back again. As you said, in the build process is including it's own temporary file, then I push to my repo and later clone it locally and then build.

**Comment by binpath** [comment]
One question, why the cap project is not possible to see the ui5 module in the HTML5 app Content provider of the launchpad service. 
I have added the configuration of FLP in the manifest.yml like Marius does, but i think i'm missing a step.

kind regards, 
@binpath

**Comment by jung-thomas** [comment]
I'm sorry that's well outside my area of expertise or the scope of this tutorial.  I think it might be best to ask that question in general in the SAP Community Q&A
